tools/mull-cxx-frontend/src/ASTMutationsSearchVisitor.cpp

```diff
@@ -0,0 +1,215 @@
+      std::unique_ptr<BinaryMutation> binaryMutator =
+          std::make_unique<BinaryMutation>(std::get<2>(mutation));
+      recordMutationPoint(std::get<1>(mutation),
+                          std::move(binaryMutator),
+                          binaryOperator,
+                          binaryOperator->getOperatorLoc());
+    }
+  }
+  return true;
+}
+
+bool ASTMutationsSearchVisitor::VisitCallExpr(clang::CallExpr *callExpr) {
+  if (callExpr->getType() == _context.VoidTy &&
+      mutationMap.isValidMutation(mull::MutatorKind::CXX_RemoveVoidCall)) {
+    std::unique_ptr<RemoveVoidMutation> removeVoidMutator = std::make_unique<RemoveVoidMutation>();
+    recordMutationPoint(mull::MutatorKind::CXX_RemoveVoidCall,
+                        std::move(removeVoidMutator),
+                        callExpr,
+                        ClangCompatibilityStmtGetBeginLoc(*callExpr));
+  }
+
+  if (callExpr->getType() == _context.IntTy &&
+      mutationMap.isValidMutation(mull::MutatorKind::CXX_ReplaceScalarCall)) {
+    std::unique_ptr<ReplaceScalarCallMutation> mutator =
+        std::make_unique<ReplaceScalarCallMutation>(callExpr);
+    recordMutationPoint(mull::MutatorKind::CXX_ReplaceScalarCall,
+                        std::move(mutator),
+                        callExpr,
+                        ClangCompatibilityStmtGetBeginLoc(*callExpr));
+  }
+
+  return true;
+}
+
+bool ASTMutationsSearchVisitor::VisitVarDecl(clang::VarDecl *D) {
+  /// Function parameter declarations are not mutated.
+  if (clang::dyn_cast_or_null<clang::ParmVarDecl>(D)) {
+    return true;
+  }
+
+  if (mutationMap.isValidMutation(mull::MutatorKind::CXX_InitConst) &&
+      (D->getType() == _context.IntTy || D->getType() == _context.FloatTy ||
+       D->getType() == _context.DoubleTy)) {
+    std::unique_ptr<ReplaceNumericInitAssignmentMutation> mutator =
+        std::make_unique<ReplaceNumericInitAssignmentMutation>(D);
+    recordMutationPoint(mull::MutatorKind::CXX_InitConst,
+                        std::move(mutator),
+                        D->getInit(),
+                        D->getInit()->getExprLoc());
+    return true;
+  }
+  return true;
+}
+
+void ASTMutationsSearchVisitor::recordMutationPoint(mull::MutatorKind mutatorKind,
+                                                    std::unique_ptr<ASTMutation> mutation,
+                                                    clang::Stmt *stmt,
+                                                    clang::SourceLocation location) {
+  if (sourceManager.isInSystemHeader(location)) {
+    return;
+  }
+  if (sourceManager.isInSystemMacro(location)) {
+    return;
+  }
+  std::string sourceFilePath = sourceManager.getFilename(location).str();
+  if (sourceFilePath.empty()) {
+    /// we reach here because of asserts()
+    /// TODO: maybe there are more cases.
+    /// assert(0);
+    return;
+  }
+  if (sourceFilePath.find("include/gtest") != std::string::npos) {
```

Comment:
I guess some copy-paste? :)

Reply:
Certainly not. It helps a lot during testing against a bigger codebase. For any GoogleTest-based project, it is very useful for now. I would really prefer to keep this until we have the filters in place. 

The same comment about the `main()` function below.